### PR TITLE
feat: [CDS-83684]: Added migration to populate Settings for HELM_STEADY_STATE_CHECK_1_16 enabled customers

### PR DESCRIPTION
**Pull request by GoelPratyush99**
## Describe your changes
Added migration to populate Settings for HELM_STEADY_STATE_CHECK_1_16 enabled customers

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55288)

### CODE REVIEW

**Comment by GoelPratyush99** [comment]
trigger uts

**Review by alebuh — DISMISSED** [review]

**Review by markoharness — APPROVED** [review]

**Review by TarunUba1 — DISMISSED** [review]
maybe add test around running migration

**Review by GoelPratyush99 — COMMENTED** [review]
@TarunUba1  They are part of this file 120-ng-manager/src/test/java/io/harness/ng/core/migration/PopulateSettingsForHelmSteadyStateCheckFFMigrationTest.java

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55288#-:-Nisi1W305-SkKlulIMx:b-yuacwq)* status: 4 file(s) to review.

**Review by abosii-harness — APPROVED** [review]

**Review by acasian-harness — APPROVED** [review]

**Comment by GoelPratyush99** [comment]
trigger uts

File: 120-ng-manager/src/main/java/io/harness/ng/core/migration/background/PopulateSettingsForHelmSteadyStateCheckFFMigration.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  @Inject private NGFeatureFlagHelperService featureFlagService;
+  @Inject private SettingRepository settingRepository;
+  private static final String DEBUG_LOG = "[PopulateSettingsForHelmSteadyStateCheckFFMigration]: ";
+
+  @Override
+  public void migrate() {
+    log.info(DEBUG_LOG + "Start migrating HELM_STEADY_STATE_CHECK_1_16 FF enabled accounts to account level settings");
+    String settingIdentifier = SettingIdentifiers.ENABLE_STEADY_STATE_FOR_JOBS_KEY_IDENTIFIER;
+    String settingTrueValue = "true";
+    AtomicInteger iterationCounter = new AtomicInteger();
```
Comment (abhinavsingh1196) [inline 120-ng-manager/src/main/java/io/harness/ng/core/migration/background/PopulateSettingsForHelmSteadyStateCheckFFMigration.java:45]: why do you need atomic integer? :)

**Review by abhinavsingh1196 — APPROVED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/core/migration/background/PopulateSettingsForHelmSteadyStateCheckFFMigration.java (hunk 1 above)
Comment (GoelPratyush99) [reply 120-ng-manager/src/main/java/io/harness/ng/core/migration/background/PopulateSettingsForHelmSteadyStateCheckFFMigration.java:45]: Just to calculate the number of settings we have created and log it

**Review by TarunUba1 — APPROVED** [review]

**Comment by TarunUba1** [comment]
trigger review

**Pull request merged**